aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/UpdateHandler.java

```diff
@@ -0,0 +1,118 @@
+                    DescribeAccountAuditConfigurationRequest.builder().build(),
+                    iotClient::describeAccountAuditConfiguration);
+        } catch (IotException e) {
+            throw Translator.translateIotExceptionToCfn(e);
+        }
+        logger.log(String.format("Called DescribeAccountAuditConfiguration for %s.", accountId));
+
+        // If RoleArn is null, the configuration must've been deleted. The RoleArn can never be nullified.
+        if (StringUtils.isEmpty(describeResponse.roleArn())) {
+            return ProgressEvent.failed(request.getDesiredResourceState(), callbackContext,
+                    HandlerErrorCode.NotFound,
+                    "The configuration for your account has not been set up or was deleted.");
+        }
+
+        Map<String, AuditCheckConfiguration> checkConfigurationsForUpdate =
+                buildCheckConfigurationsForUpdate(model, describeResponse);
+
+        Map<String, AuditNotificationTarget> notificationTargetConfigurationsForUpdate =
+                buildNotificationTargetConfigurationsForUpdate(model);
+
+        UpdateAccountAuditConfigurationRequest updateRequest = UpdateAccountAuditConfigurationRequest.builder()
+                .auditCheckConfigurations(checkConfigurationsForUpdate)
+                .auditNotificationTargetConfigurationsWithStrings(notificationTargetConfigurationsForUpdate)
+                .roleArn(model.getRoleArn())
+                .build();
+        try {
+            proxy.injectCredentialsAndInvokeV2(
+                    updateRequest, iotClient::updateAccountAuditConfiguration);
+        } catch (IotException e) {
+            throw Translator.translateIotExceptionToCfn(e);
+        }
+
+        logger.log(String.format("Updated AccountAuditConfiguration for %s.", accountId));
+        return ProgressEvent.defaultSuccessHandler(request.getDesiredResourceState());
+    }
+
+    Map<String, AuditCheckConfiguration> buildCheckConfigurationsForUpdate(
+            ResourceModel model,
+            DescribeAccountAuditConfigurationResponse describeResponse) {
+
+        // The UpdateAccountAuditConfiguration API has a PATCH behavior, so we can't just
+        // translate the map from the model.
+        // A customer can enable a check out of band that this template doesn't mention.
+        // We create the map in 2 steps. First, create a map with all supported checks disabled.
+        // Second, copy all the entries from the model.
+        Map<String, AuditCheckConfiguration> checkConfigurationsForUpdate = new HashMap<>();
+        describeResponse.auditCheckConfigurations().keySet().forEach(k ->
+                checkConfigurationsForUpdate.put(k, AuditCheckConfiguration.builder().enabled(false).build()));
+        model.getAuditCheckConfigurations().forEach((key, value) ->
+                checkConfigurationsForUpdate.put(key, Translator.translateCheckConfigurationFromCfnToIot(value)));
```

Comment:
This means that any out of band changes will be overriden. Does this match cloudformation's guidance for out of band changes?

Reply:
(Discussed offline and agreed) CFN doesn't provide guidance on this, they leave this up to services. We'll stay consistent across the Device Defender resources and will treat the template as the authoritative desired configuration, overriding changes made out-of-band.